### ISSUE TITLE
Inventory not being rendered

### ISSUE THREAD

**Issue by theviper121** [minetest/minetest#9640]
##### Summary
Inventory not being rendered when it is created on the same server step the form using it is send.
It seems the client is not yet aware of the newly created inventory.
##### Steps to reproduce
Create a detached inventory and send formspec to client on same step to get:
GUIFormSpecMenu::parseList(): The inventory list ##### @ ####### doesn't exist

Seems the issue was introduced by commit 784382253cbf31d6bc7e8724b3a4d200a76eb7b3
In file src/gui/guiFormSpecMenu.cpp

**Comment by SmallJoker** [comment]
Inventory lists and inventory formspecs are sent very closely together:

https://github.com/minetest/minetest/blob/aa3cf400e2ca4096517e926dbae5b6337b2f1876/src/server.cpp#L1005-L1009

Does this happen using the default settings (i.e. rename "minetest.conf")?

> GUIFormSpecMenu::parseList(): The inventory list ##### @ ####### doesn't exist

Which lists? Please provide the each different warning log line.

**Comment by Desour** [comment]
I can reproduce, sometimes.
In the old formspec code, the inventory list tried again and again, each draw step, to get the inventorylist. Now it doesn't anymore. Before the formspec changes, there were probably also warnings.

<details><summary>Code I've used for testing</summary>

```lua
local detached_test_inv_exists = false
local function show_inv_formspec(name)
	print("opening inv test formspec...")
	local inv_name = "fs_test_detached_inv"
	if not detached_test_inv_exists then
		local inv = minetest.create_detached_inventory(inv_name, {
			allow_move = function(inv, from_list, from_index, to_list, to_index, count, player)
				return count
			end,
			allow_put = function(inv, listname, index, stack, player)
				return stack:get_count()
			end,
			allow_take = function(inv, listname, index, stack, player)
				return stack:get_count()
			end,
			--~ on_move = function(inv, from_list, from_index, to_list, to_index, count, player),
			on_put = function(inv, listname, index, stack, player)
				minetest.remove_detached_inventory(inv_name)
				detached_test_inv_exists = false
			end,
			--~ on_take = function(inv, listname, index, stack, player),
		})
		inv:set_size("baum", 6)
		detached_test_inv_exists = true
	end
	--~ minetest.remove_detached_inventory(inv_name)
	minetest.show_formspec(name, "fs_test_inv", "formspec_version[3]"..
		"size["..(9 + 7 * 0.25)..",10]"..
		"list[detached:"..inv_name..";baum;0.5,0.5;3,2;0]"..
		"list[current_player;main;0.5,3.25;8,4;]"
	)
	print("finished.")
end

minetest.register_chatcommand("fs", {
	params = "",
	description = "Show the test formspec.",
	privs = {},
	func = function(name, param)
		show_inv_formspec(name)
		return true
	end,
})
```
</details>

**Comment by oilboi** [comment]
Wouldn't the simple solution be to send the inventory before the formspec?

**Comment by KaylebJay** [comment]
This issue arises on the server Tunnelers' Abyss with the mod 'inventorybags' on clients newer than 5.1.1, to confirm.

**Comment by rubenwardy** [comment]
This happens when granting creative in a non-creative game

**Comment by Desour** [comment]
Fix: #9700
